packages/domain/lib/usecase/analytics_usecase.dart

```diff
@@ -0,0 +1,13 @@
+import 'package:domain/services/analytics_service.dart';
+import 'package:domain/usecase/usecase.dart';
+
+class LogAnalyticsEventUseCase implements UseCaseParams<String, Future<void>> {
+  final AnalyticsService _analyticsService;
+
+  LogAnalyticsEventUseCase(this._analyticsService);
+
+  @override
+  Future<void> call(String params) async {
+    await _analyticsService.logEvent(params);
+  }
```

Comment:
u can do in one line

`Future<void> call(String params) async => _analyticsService.logEvent(params);`

there's no need in await